### PR TITLE
Implement 'Nodes Network' test for GKE

### PR DESCRIPTION
**Pull request by mbforbes**
- Switch all providers for this test to use the Kubernetes master IP instead of the DNS name.
  - @justinsb does this look good for AWS?
  - @marekbiskup these are your TODOs, does this look all right?
- Remove `KUBE_MASTER` from GKE's `util.sh` because it's wrong and not lookup-able
- _I think_ fix a typo in `hack/ginkgo-e2e.sh` with the defaulting `:`
  - @ixdy can you confirm?

If any of the above would like to jump on as a reviewer, I'd be flattered, but I can also find one after you guys chime in. Thank you!

### CODE REVIEW

File: hack/ginkgo-e2e.sh [hunk 1]
```diff
@@ -89,7 +89,7 @@ fi
 export PATH=$(dirname "${e2e_test}"):"${PATH}"
 "${ginkgo}" "${ginkgo_args[@]:+${ginkgo_args[@]}}" "${e2e_test}" -- \
   "${auth_config[@]:+${auth_config[@]}}" \
-  --host="https://${KUBE_MASTER_IP-}" \
+  --host="https://${KUBE_MASTER_IP:-}" \
```
Comment (ixdy) [inline hack/ginkgo-e2e.sh]: Assuming you want this to be `--host="https://"` when `$KUBE_MASTER_IP` is null, then yes, this is correct.

But maybe this should instead be `--host=${KUBE_MASTER_IP:+"https://${KUBE_MASTER_IP}"}` or `${KUBE_MASTER_IP:+"--host=https://${KUBE_MASTER_IP}"}` ?

Comment (ixdy) [inline hack/ginkgo-e2e.sh]: I think as it's currently written, there's no real difference between `:-` and `-` - it will either give you the value of `$KUBE_MASTER_IP` or the empty string.

Comment (mbforbes) [inline hack/ginkgo-e2e.sh]: It seems like the simplest thing is to keep it as is (and have it be `https://` when `KUBE_MASTER_IP` isn't set). That way we can assume that the value of the var begins with `https://` and don't have to add special casing. Then, we could consider doing a cleanup change in a future PR. Does this sound all right?

**Comment by mbforbes** [comment]
@k8s-bot test this please

**Comment by mbforbes** [comment]
@ixdy / @marekbiskup / @justinsb any comments, or does this LGTY?

**Comment by mbforbes** [comment]
Friendly ping @ixdy / @marekbiskup / @justinsb I'd love to get this in ASAP

File: test/e2e/resize_nodes.go [hunk 2]
```diff
@@ -329,22 +329,8 @@ func performTemporaryNetworkFailure(c *client.Client, ns, rcName string, replica
...
-	destination := testContext.CloudConfig.MasterName
-	if providerIs("aws") {
-		// This is the (internal) IP address used on AWS for the master
-		// TODO: Use IP address for all clouds?
-		// TODO: Avoid hard-coding this
-		destination = "172.20.0.9"
-	}
-
-	iptablesRule := fmt.Sprintf("OUTPUT --destination %s --jump DROP", destination)
+	iptablesRule := fmt.Sprintf("OUTPUT --destination %s --jump DROP",
```
Comment (ixdy) [inline test/e2e/resize_nodes.go]: if I'm understanding this code correctly, there are potentially cases where we pass an empty string to --destination (like if KUBE_MASTER_IP is not set). is that valid behavior?

Comment (mbforbes) [inline test/e2e/resize_nodes.go]: I think it's simply a different assumption. We used to assume that `KUBE_MASTER` is set; now, we assume that `KUBE_MASTER_IP` is set (which works for more providers).

Did you want me to add a check at the beginning of the test for this? It would be a fail-safe for future situations / providers that don't have `KUBE_MASTER_IP` set, so would it be all right to do in a future PR if so?

Comment (ixdy) [inline test/e2e/resize_nodes.go]: it'd probably be good to verify that `KUBE_MASTER_IP` is set (and fail the test if not), but sure, that can be a separate PR.

**Comment by ixdy** [comment]
LGTM

**Comment by wojtek-t** [comment]
This PR broke our e2e tests - the "recreates pods scheduled on the unreachable minion node AND allows scheduling of pods on a minion after it rejoins the cluster" test is constantly failing after this PR.

I'm going to revert this PR so fix Jenkins.
